src/Controller/WebexController.php

```diff
@@ -0,0 +1,103 @@
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Routing\RouterInterface;
+
+class WebexController extends AbstractController
+{
+    public function __construct(
+        private string $webexClientId,
+        private string $webexClientSecret,
+        private RouterInterface $router,
+    ) {
+    }
+
+    /**
+     * Ask for Webex authentication and store the AccessToken in Session.
+     */
+    #[Route('/auth/webex', name: 'webex_authenticate', methods: ['GET'])]
+    public function authenticate(Request $request, WebexApplication $webexApplication, UserExtractor $userExtractor): Response
+    {
+        $session = $request->getSession();
+
+        $provider = new WebexProvider([
+            'clientId' => $this->webexClientId,
+            'clientSecret' => $this->webexClientSecret,
+            'redirectUri' => $this->router->generate('webex_authenticate', [], RouterInterface::ABSOLUTE_URL),
+        ]);
+
+        if ($request->query->has('error')) {
+            if ('invalid_scope' === $request->query->get('error')) {
+                throw new AuthenticationException(WebexApplication::APPLICATION_CODE, 'Cannot get the required permissions, are you Webex admin?');
+            }
+
+            return $this->redirectToRoute('homepage');
+        }
+
+        if (!$request->query->has('code')) {
+            // If we don't have an authorization code then get one
+            $options = [
+                'scope' => [
+                    'spark:kms', // This scope is required to give your integration permission to interact with encrypted content (such as messages).
+                    // 'spark:people_read',
+                    //
+                    'spark-admin:people_read',
+                    // 'spark:teams_read',
+                ],
+            ];
+            $authUrl = $provider->getAuthorizationUrl($options);
+
+            $session->set(WebexApplication::SESSION_KEY_STATE, $provider->getState());
+
+            return new RedirectResponse($authUrl);
+        }
+
+        // Check given state against previously stored one to mitigate CSRF attack
+        if (empty($request->query->get('state')) || ($request->query->get('state') !== $session->get(WebexApplication::SESSION_KEY_STATE))) {
+            $session->remove(WebexApplication::SESSION_KEY_STATE);
+
+            throw new AuthenticationException(WebexApplication::APPLICATION_CODE, 'Invalid OAuth state.');
+        }
+
+        try {
+            // Try to get an access token (using the authorization code grant)
+            /** @var AccessToken $token */
+            $token = $provider->getAccessToken('authorization_code', [
+                'code' => $request->query->get('code'),
+            ]);
+
+            $appToken = new ApplicationToken($token->getToken());
+            $admin = $userExtractor->getMe($token->getToken());
+        } catch (\Exception $e) {
```

Comment:
Please, never use `\Exception`. It makes debuging much harder. Could you narrow the exception class?

Reply:
We call an external library method and which is badly documented, in our case AuthenticationException is catched by a global listener and display a nice page to the user: we did it on purpose.

Reply:
So if a network error occurs (for example), we display a wrong message to the end user?